web/core/store/workspace/index.ts

```diff
@@ -214,18 +214,29 @@ export class WorkspaceRootStore implements IWorkspaceRootStore {
     key: string,
     data: Partial<IWorkspaceSidebarNavigationItem>
   ) => {
+    // Store the data before update to use for reverting if needed
+    const beforeUpdateData = this.navigationPreferencesMap[workspaceSlug]?.[key];
```

Comment:
Can you use `clone` or `closeDeep` from lodash over here? This just keeps the reference of the object and the logic might not work properly in this case.